src/coreclr/nativeaot/Runtime/arm/InteropThunksHelpers.S

```diff
@@ -46,6 +46,7 @@ LEAF_ENTRY RhGetCommonStubAddress, _TEXT
           bx            lr
 LEAF_END RhGetCommonStubAddress, _TEXT
 
+#ifndef FEATURE_EMULATED_TLS
 //
 // IntPtr RhGetCurrentThunkContext()
```

Comment:
Is there a reason for why this is implemented in arch specific assembly in the first place?

Reply:
I don't think so. This just matches what other platforms were doing but I assume we can remove the assembly implementations everywhere.

Reply:
I think it would be best to delete all this unnecessary asm code.